h2olog

```diff
@@ -93,7 +93,11 @@ quic_bpf = """
 #define MAX_HEADER_VALUE_LEN 128
 #define TOKEN_PREVIEW_LEN 8
 
-int sprintf(char * restrict str, const char * restrict format, ...);
+#include <linux/kernel.h>
+
+#define INIT_EVENT_NAME(event) do { \\
+        sprintf((event).type, __func__ + (sizeof("trace_") -1)); \\
```

Comment:
As a convention, strcpy should be used for copying strings. Use of sprintf is discouraged as the function expects more argument depending on the format being given (and therefore providing a variable format is not considered a good idea).

I also think that we have buffer overflow here. I presume that the longest string given as the second argument would be "quicly__stream_data_blocked_receive" which is 35 bytes long, whereas `(event).type` is 32 bytes long.